src/teams.rs

```diff
@@ -106,17 +106,36 @@ impl Data {
         let mut wgs = Vec::new();
         let mut project_groups = Vec::new();
         self.teams
-            .into_iter()
+            .iter()
             .filter(|team| team.website_data.is_some())
             .filter(|team| team.subteam_of.as_ref() == Some(&main_team.name))
-            .for_each(|team| match team.kind {
-                TeamKind::Team => subteams.push(team),
-                TeamKind::WorkingGroup => wgs.push(team),
-                TeamKind::ProjectGroup => project_groups.push(team),
+            .for_each(|subteam| match subteam.kind {
+                TeamKind::Team => {
+                    subteams.push(subteam.clone());
+
+                    subteams.extend(
+                        self.teams
+                            .iter()
+                            .filter(|team| team.subteam_of.as_ref() == Some(&subteam.name))
+                            .filter(|team| matches!(team.kind, TeamKind::Team))
+                            .filter(|team| team.website_data.is_some())
+                            .cloned(),
+                    );
+                }
+                TeamKind::WorkingGroup => wgs.push(subteam.clone()),
+                TeamKind::ProjectGroup => project_groups.push(subteam.clone()),
                 _ => {}
```

Comment:
Can't we put this into a function which can eventually call itself? Like that we wouldn't have trouble if have super nested subteams/subgroups.

Reply:
I can try to impl an recursive or iterative approach. Earlier I've attempted sth. like that it but since the teams are stored in a `Vec`, not a `HashMap` or graph, I had to linearly look through the `Vec` again and again, yielding a bad algorithmic complexity. Let me try again